src/viam/sdk/common/version.cpp.in

```diff
@@ -0,0 +1,9 @@
+#include <viam/sdk/common/version.hpp>
+
+namespace viam {
+namespace sdk {
+
+const char* const k_version = "cpp;@VIAMCPPSDK_SDK_VERSION@;@VIAMCPPSDK_API_PROTO_LABEL@";
```

Comment:
- I don't think this string belongs as part of the public interface of the SDK, since it is entirely an implementation detail. Consider creating a `viam/sdk/common/private/version_metadata.[hpp,cpp]` instead.
- I don't think you need the indirection through a pointer: `constexpr char k_version_metadata[] = "..."`.
- We probably someday will want a public `version.hpp` which contains the major/minor/patch versions broken out, along with some convenience methods.

Reply:
Cool, will make this private, remove the pointer-ness of it, and create a ticket for an eventual public `version.hpp`!

Reply:
ticket: https://viam.atlassian.net/browse/RSDK-8343

Reply:
If we're making it private we might consider whether we want to skip the header/source split entirely, see eg 
https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/common/utils.hpp#L20
making it `const` gives it the right linkage anyway, and if it's only ever included in a `.cpp` file I don't think we gain anything from breaking it off compilation-wise. And then I think it's a tossup whether we want it `std::string` or `char` ptr/array

Reply:
I wish we had `string_view` because a `constexpr string_view` is really the best thing for this. I don't care too much if we want to make it a std::string, though I don't think it buys us much over a char array in this particular case. I agree that, as a private header, we can certainly avoid the need for a separate .cpp. I would like to see the file renamed from `version.hpp` to `version_metadata.hpp` though.

Reply:
There is `boost::core::string_view` although it may be a bit silly to use just for this thing

Reply:
Yes, but if we want it, I'd like to either 1) go to C++17 as the language standard of record for the project and then just use it, or 2) polyfill it in to use either C++17 when built that way or boost when built for C++14. So, let's just defer that and stay with either `std::string` or `char` arrray for now.

Reply:
renamed to `version_metadata.hpp`, sticking with `char` now for simplicity. Do we want to get rid of the `.cpp` file entirely? It makes sense to me but I don't have a particular opinion either way, happy to defer here.